### PR TITLE
Move enums from tags_int.hpp to tags.hpp

### PR DESCRIPTION
**Pull request by kevinbackhouse**
This moves the declarations of enums `IfdId` and `SectionId` from `src/tags_int.hpp` to `include/exiv2/tags.hpp`. I think it makes the code a bit cleaner because now the `TagInfo` struct can use those enums rather than `int`.

But maybe there's a reason why `IfdId` and `SectionId` shouldn't be included in the public interface that I'm not aware of? If so, I'll close this PR.

### CODE REVIEW

**Review by piponazo — DISMISSED** [review]
I do not have anything against moving that enum to the API, and it makes sense to have it there so that `TagInfo` can reuse it. 👍

File: include/exiv2/tags.hpp [hunk 1]
```diff
@@ -47,18 +47,192 @@ struct EXIV2API GroupInfo::GroupName {
   std::string g_;  //!< Group name
 };
 
+//! Type to specify the IFD to which a metadata belongs
+enum IfdId {
```
[the remark this answers is not in the document]
Comment (kevinbackhouse) [reply include/exiv2/tags.hpp:51]: good suggestion!

[the remark this answers is not in the document]
Comment (kevinbackhouse) [reply include/exiv2/tags.hpp]: I added a commit that does this. It's a big diff! I used a script to make most of the changes.

**Comment by neheb** [comment]
```
--- ./src/image_int.hpp	(original)
+++ ./src/image_int.hpp	(reformatted)
@@ -65,7 +65,7 @@
   explicit binaryToStringHelper(const Slice<T> myBuf) noexcept : buf_(myBuf) {
   }
-  friend std::ostream& operator<< <T>(std::ostream& stream, const binaryToStringHelper<T>& binToStr);
+  friend std::ostream& operator<<<T>(std::ostream& stream, const binaryToStringHelper<T>& binToStr);
   // the Slice is stored by value to avoid dangling references, in case we
   // invoke:
```

This is a difference between clang-format-12 and 14. Needs to be updated IMO.

**Review by piponazo — APPROVED** [review]
Thanks a lot! Specially for the additional effort for the final commit 👏

**Comment by piponazo** [comment]
> This is a difference between clang-format-12 and 14. Needs to be updated IMO.

Few months ago when I introduced the clang-format workflow we discussed about this and we decided to use the clang-version coming with `ubuntu-latest`. However, if this is bothering you guys, I do not any objection to upgrade to clang-format-14. It can be done by just updating this file: `.github/workflows/on_push_clang_format.yml`.  More information about the action can be found here:

https://github.com/DoozyX/clang-format-lint-action

**Comment by kevinbackhouse** [comment]
@piponazo @neheb: Regarding the clang-format version, I think the best choice is whatever people are most likely to have installed on their computers. I'm running Ubuntu 22.04, which comes with clang 14, but I expect most people are using Macs or Windows, and I don't know which version is the default on those platforms.

I can easily use a docker container to run version 12 if necessary, so I'm happy to work around this if 12 is the most popular choice.

**Comment by jim-easterbrook** [comment]
> But maybe there's a reason why `IfdId` and `SectionId` shouldn't be included in the public interface that I'm not aware of? If so, I'll close this PR.

I just happened across this, making `IfdId` internal 11 years ago: https://dev.exiv2.org/issues/721

**Pull request merged**